js/script.js

```diff
@@ -0,0 +1,787 @@
+
+      data.operation.push(symbol);
+      data.formula.push(formula);
+    } else if (button.name == "exp") {
+      symbol = "exp(";
+      formula = button.formula + "(";
+
+      data.operation.push(symbol);
+      data.formula.push(formula);
+    } else {
+      symbol = button.symbol + "(";
+      formula = button.formula;
+      data.operation.push(symbol);
+      data.formula.push(formula);
+    }
+  } else if (button.type == "key") {
+    if (button.name == "clear") {
+      data.operation = [];
+      data.formula = [];
+      updateOutputresult(0);
+    } else if (button.name == "delete") {
+      data.operation.pop();
+      data.formula.pop();
+    } else if (button.name == "deg") {
+      RADIAN = false;
+      angleToggle();
+    }
+  } else if (button.type == "calculate") {
+    formula_str = data.formula.join("");
+    let POWER_SEARCH_RESULT = search(data.formula, POWER);
+    let FACTORIAL_SEARCH_RESULT = search(data.formula, FACTORIAL);
+    const BASES = powerbasegetter(data.formula, POWER_SEARCH_RESULT);
+    const NUMBERS = factorialnumgetter(data.formula, FACTORIAL_SEARCH_RESULT);
+
+    BASES.forEach((base) => {
+      let toreplace = base + POWER;
+      let replacement = "Math.pow(" + base + ",";
+
+      formula_str = formula_str.replace(toreplace, replacement);
+    });
+
+    NUMBERS.forEach((number) => {
+      formula_str = formula_str.replace(number.toReplace, number.replacement);
+    });
+
+    try {
+      result = eval(formula_str);
+      var decCount = decimalCount(result);
+      if (decCount > 4) {
+        result = result.toPrecision(4);
+      }
+    } catch (error) {
+      if (error instanceof SyntaxError) {
+        result = "SyntaxError";
+        updateOutputresult(result);
+        return;
+      }
+    }
+
+    ans = result;
+    data.operation = [result];
+    data.formula = [result];
+    updateOutputresult(result);
+    return;
+  }
+  updateOutputOperation(data.operation.join(""));
+}
+
+function factorialnumgetter(formula, FACTORIAL_SEARCH_RESULT) {
+  let numbers = [];
+
+  FACTORIAL_SEARCH_RESULT.forEach((fact_index) => {
+    let number = [];
+    let prev_idx = fact_index - 1;
+
+    while (prev_idx >= 0) {
+      let is_operator = false;
+
+      OPERATORS.forEach((OPERATOR) => {
+        if (formula[prev_idx] == OPERATOR) {
+          is_operator = true;
+        }
+      });
+
+      if ((is_operator)) {
+        break;
+      }
+
+      number.unshift(formula[prev_idx]);
+      prev_idx--;
+    }
+
+    let number_str = number.join("");
+    const factorial = "factorial(",
+      close_paren = ")";
+    let toreplace = number_str + FACTORIAL;
+    let replacement = factorial + number_str + close_paren;
+    numbers.push({
+      toReplace: toreplace,
+      replacement: replacement,
+    });
+  });
+  return numbers;
+}
+
+function powerbasegetter(formula, POWER_SEARCH_RESULT) {
+  let powers_base = [];
+
+  POWER_SEARCH_RESULT.forEach((power_index) => {
```

Comment:
Array index can be accessed as second argument of callback function in forEach method.
Eg, 
`
Array.forEach((element, index) => { 
/*body*/
})
`